### PR TITLE
DOC: Add more examples for `np.c_`

### PR DESCRIPTION
**Pull request by flying-sheep**
I mostly use that object to bind 1D arrays as columns, so I added an example for that use case.

### CODE REVIEW

File: numpy/lib/index_tricks.py [hunk 1]
```diff
@@ -468,6 +468,10 @@ class CClass(AxisConcatenator):
 
     Examples
     --------
+    >>> np.c_[np.array([1,2,3]]), np.array([4,5,6])]
```
Comment (eric-wieser) [inline numpy/lib/index_tricks.py]: Extra `]` here

**Review by eric-wieser — CHANGES_REQUESTED** [review]

**Comment by eric-wieser** [comment]
Wait, if this passed does that mean we're not running doctests, @charris?

**Comment by seberg** [comment]
@eric-wieser yup, we have not been running doc tests for years.

**Comment by eric-wieser** [comment]
@seberg: What was the rationale behind their removal? (And the commit/issue that removed them)

**Review by flying-sheep — COMMENTED** [review]
ouch, sorry about that blunder

**Comment by eric-wieser** [comment]
Commits need squashing, and the message should start with `DOC: `

**Comment by eric-wieser** [comment]
And while you're there, might be nice to add a `See Also` section for `column_stack`

**Comment by flying-sheep** [comment]
sure, i’ll add the section.

can’t just just squash from github?

**Comment by eric-wieser** [comment]
Looks good to me

**Review by eric-wieser — APPROVED** [review]

**Comment by eric-wieser** [comment]
Little concerned about the smart quotes in the commit message. @charris - is this an issue?

**Comment by charris** [comment]
@eric-wieser doctests are hard to maintain, harder to make both python2 and python3 compatible, depend on the details of formatting, which may change, and generally don't make for good tests. That isn't to say they aren't somewhat maintained as examples, but we don't use them for testing purposes.

**Comment by eric-wieser** [comment]
>  harder to make both python2 and python3 compatible,

Perhaps we should just enable them on python3 then?

> generally don't make for good tests

They make for good tests for whether the documentation is up to date :)

**Comment by charris** [comment]
What are the smart quotes?

**Comment by eric-wieser** [comment]
> DOC: Added “See Also” section for c_

vs

> DOC: Added "See Also" section for c_

Just checking that we have tooling that supports non-ascii characters in commit messages (I assume those are utf8-encoded)

**Comment by charris** [comment]
IIRC, MIcrosoft uses latin-1 or UTF-16, neither of which is compatible with UTF-8, so it is probably best to stick with the "dumb" quotes.

**Comment by eric-wieser** [comment]
Apparently git encodes in utf8 automatically, so I think we should be good

**Comment by flying-sheep** [comment]
great!

**Comment by ev-br** [comment]
Re doctests: numpy might want to reuse https://github.com/scipy/scipy/blob/master/tools/refguide_check.py

**Pull request merged**